frontend/src/features/qa/page-qa-toolbar.ts

```diff
@@ -0,0 +1,454 @@
+          })}
+          aria-checked=${approved}
+          ?disabled=${disabled}
+          @click=${() =>
+            this.submitReview({ approved: approved ? null : true })}
+        >
+          <sl-icon name="hand-thumbs-up" label=${msg("Approve")}></sl-icon>
+        </button>
+        <button
+          role="checkbox"
+          class=${classMap({
+            comment: true,
+            active: commented,
+          })}
+          aria-checked=${commented}
+          ?disabled=${disabled}
+          @click=${() => (this.showComments = true)}
+        >
+          <sl-icon name="chat-square-text" label=${msg("Comment")}></sl-icon>
+        </button>
+        <button
+          class=${classMap({
+            rate: true,
+            reject: true,
+            active: rejected,
+          })}
+          aria-checked=${rejected}
+          ?disabled=${disabled}
+          @click=${() =>
+            this.submitReview({ approved: rejected ? null : false })}
+        >
+          <sl-icon name="hand-thumbs-down" label=${msg("Reject")}></sl-icon>
+        </button>
+      </fieldset>
+
+      <btrix-dialog
+        label=${msg("Page Review Comments")}
+        ?open=${this.showComments}
+        @sl-hide=${() => (this.showComments = false)}
+        @sl-after-hide=${() => this.fetchPage()}
+      >
+        ${keyed(this.showComments, this.renderComments())}
+        <p
+          slot="footer"
+          class="mb-2 rounded border border-slate-200 bg-slate-50 p-2 text-left text-neutral-500"
+        >
+          <sl-tag size="small" variant="primary" class="mr-1"
+            >${msg("Beta Notice")}</sl-tag
+          >
+          ${msg(
+            "We may analyze anonymized text from this comment to improve the beta QA feature.",
+          )}
+        </p>
+        <sl-button
+          slot="footer"
+          size="small"
+          variant="primary"
+          @click=${() => this.dialog.submit()}
+        >
+          ${msg("Submit Comment")}
+        </sl-button>
+      </btrix-dialog>
+    `;
+  }
+
+  private renderComments() {
+    const comments = this.page?.notes || [];
+    return html`
+      ${when(
+        comments.length,
+        () => html`
+          <btrix-details>
+            <span slot="title"
+              >${msg(str`Comments (${comments.length.toLocaleString()})`)}</span
+            >
+            ${when(
+              this.page?.notes,
+              (notes) => html`
+                <ul>
+                  ${notes.map(
+                    (comment) =>
+                      html`<li class="mb-3">
+                        <div
+                          class="flex items-center justify-between rounded-t border bg-neutral-50 text-xs leading-none text-neutral-600"
+                        >
+                          <div class="p-2">
+                            ${msg(
+                              str`${comment.userName} commented on ${new Date(comment.created + "Z").toLocaleDateString()}:`,
```

Comment:
Would remove this colon?  It's grammatically correct... but looks kinda weird in a title?
```suggestion
                              str`${comment.userName} commented on ${new Date(comment.created + "Z").toLocaleDateString()}`,
```